packages/twenty-server/src/engine/core-modules/auth/services/token.service.ts

```diff
@@ -91,9 +94,33 @@ export class TokenService {
       );
     }
 
+    const workspaceIdNonNullable = workspaceId
+      ? workspaceId
+      : user.defaultWorkspace.id;
+
+    const workspaceMemberRepository =
+      await this.twentyORMGlobalManager.getRepositoryForWorkspace<WorkspaceMemberWorkspaceEntity>(
+        workspaceIdNonNullable,
+        'workspaceMember',
+      );
+
+    const workspaceMember = await workspaceMemberRepository.findOne({
+      where: {
+        userId: user.id,
+      },
+    });
+
+    if (!workspaceMember) {
+      throw new AuthException(
+        'User is not a member of the workspace',
+        AuthExceptionCode.FORBIDDEN_EXCEPTION,
+      );
+    }
+
     const jwtPayload: JwtPayload = {
       sub: user.id,
       workspaceId: workspaceId ? workspaceId : user.defaultWorkspace.id,
+      workspaceMemberId: workspaceMember.id,
```

Comment:
are you sure we need the workspaceMemberId in the token, as we can retrieve the workspaceMember using the userId

Reply:
@martmull it means we retrieve the workspace member for each creation. Adding the workspaceId once in the token avoids some call to DB